editor/app/src/main/java/ca/ubc/eml/soiltopargraphy/editor/ui/main/MainFragment.kt

```diff
@@ -75,6 +81,28 @@ class MainFragment : Fragment() {
             activity!!.invalidateOptionsMenu()
         }
 
+        // Button to add a new flag
+        val addFlag = view.findViewById<View>(R.id.addButton)
+        addFlag.setOnClickListener {
+
+            //Add flag to centre of screen
+
+            val relativeLayout = RelativeLayout(this.context)
```

Comment:
Rather than making a new layout, just reuse [`ConstraintLayout.LayoutParams`](https://developer.android.com/reference/android/support/constraint/ConstraintLayout.LayoutParams).